engine/src/main/java/org/destinationsol/assets/audio/OggMusicManager.java

```diff
@@ -145,53 +148,47 @@ public void changeVolume(GameOptions options) {
     /**
      * Registers all music from the module of shipName
      *
-     * @param shipName Name of the ship that is loaded when creating a new game
+     * @param moduleName Name of the module of ship that is loaded when creating a new game
      */
-    public void registerModuleMusic(String shipName) {
-        Set<ResourceUrn> configUrnList = Assets.getAssetHelper().list(Json.class, "[a-zA-Z]*:playerSpawnConfig");
-
-        for (ResourceUrn configUrn : configUrnList) {
-            Json json = Assets.getJson(configUrn.toString());
-            JSONObject rootNode = json.getJsonValue();
-
-            if (rootNode.keySet().contains(shipName) && rootNode.get(shipName) instanceof JSONObject) {
-                String moduleName = configUrn.toString().split(":")[0];
-                Json musicJson;
-
-                try {
-                    musicJson = Assets.getJson(moduleName + ":musicConfig");
-                } catch (RuntimeException e) {
-                    logger.warn("Music config not found for module " + moduleName + ":musicConfig");
-                    return;
-                }
-
-                MusicConfig musicConfig = MusicConfig.load(moduleName, musicJson.getJsonValue());
-                Map<String, List<String>> musicMap = musicConfig.getMusicMap();
-
-                this.musicMap.remove(MENU_MUSIC_SET);
-                for (String music : musicMap.get(MENU_MUSIC_SET)) {
-                    registerMusic(MENU_MUSIC_SET, music);
-                }
-                for (String music : musicMap.get(GAME_MUSIC_SET)) {
-                    registerMusic(GAME_MUSIC_SET, music);
-                    logger.info("Registered " + music);
-                }
+    public void registerModuleMusic(String moduleName, final GameOptions options) {
+        if (currentlyRegisteredModule != null) {
+            if (moduleName.equals(currentlyRegisteredModule)) {
+                //Skips directly to module music
+                stopMusic();
+                playMusicTrack(this.musicMap.get(GAME_MUSIC_SET).get(2), options);
+                return;
+            } else {
+                unregisterModuleMusic();
+                currentlyRegisteredModule = null;
             }
+        }
 
-            json.dispose();
+        if (Assets.getAssetHelper().list(Json.class, moduleName + ":musicConfig").isEmpty()) {
+            logger.warn("Music config not found for module " + moduleName + ":musicConfig");
+            return;
```

Comment:
Be careful with `return` statements for error handling in large methods like this. By skipping the remainder of the code in the method you need to be sure none of it is needed to do cleanup or other steps you want to make sure happens regardless of errors.

See https://github.com/MovingBlocks/DestinationSol/issues/373#issuecomment-442121528 - you seem to have previously introduced an issue that's resolved by switching a `return` to a `break` 

While this PR seems to resolve that issue it is an indicator that the way this is written may not be quite right yet. If music config is optional for a module why are we logging a warning when not finding it? That's okay, that music just doesn't need it. 

Take a close look at your logic flow. Review that each `return` makes sense where it is as opposed to reorganizing the code in a method to where you don't need them (better if-then-else tends to help). Consider the different possibilities - is there any way to end up in a weird situation (a module that has music but no music config, or vice versa) that would run through your logic in an unexpected way?

At one point during testing (before this PR, with the prior music issue present) I somehow provoked a situation in which the game crashed, but didn't close gracefully. Usually that's an indicator that the logic flow around error handling went sideways and didn't manage to clean up after itself. I am not sure exactly how I provoked it and this PR would probably change that whole area anyway, but that's one of the sort of things to look out for :-)

Reply:
Yes the return caused errors as the default music was registered only after the module music is loaded. If no module music was loaded then the default music would never be registered. However this was fixed in the new PR as default music is loaded separately. I've also added better log messages now.